lib/active_record_host_pool/connection_adapter_mixin.rb

```diff
@@ -57,6 +57,7 @@ def _switch_connection
         log("select_db #{_host_pool_current_database}", "SQL") do
           clear_cache! if respond_to?(:clear_cache!)
           raw_connection.select_db(_host_pool_current_database)
+          @config[:database] = _host_pool_current_database if ActiveRecord::VERSION::MAJOR >= 5
```

Comment:
@grosser FYI `data_sources` and all methods based on it like `table_exists?` weren't working with the host pool gem.